packages/plugin-ext/src/main/browser/tabs/tabs-main.ts

```diff
@@ -110,6 +113,22 @@ export class TabsMainImpl implements TabsMain, Disposable {
         });
     }
 
+    waitForWidget(widget: Widget): Promise<void> {
+        const deferred = new Deferred<void>();
+        this.waitQueue.set(widget, deferred);
+
+        const timeout = setTimeout(() => {
+            deferred.resolve(); // resolve to unblock the event
```

Comment:
How are we running into this case?

Reply:
For example, if the widget has been added (somehow) to a non-main-panel tabbar. We probably still want to trigger the event on the plugin host, even if the tab API is not aware of the widget.